### PR TITLE
[JIT] ARM64 - Combine `cmp` and shift ops into a single `cmp` op

### PR DESCRIPTION
**Pull request by TIHan**
## Description

Given this example:
```asm
lsl    w1, w1, #2
cmp    w0, w1
```

Can be converted into a single `cmp`:
```asm
cmp    w0, w1, LSL #2
```

This PR will not include the `cmn` version.

### CODE REVIEW

**Comment by TIHan** [comment]
@dotnet/jit-contrib @BruceForstall @tannergooding This is ready. CI linux formatting failure is unrelated.

**Comment by TIHan** [comment]
[Diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=237366&view=ms.vss-build-web.run-extensions-tab)

File: src/coreclr/jit/codegenarm64.cpp [hunk 1]
```diff
@@ -10329,4 +10314,29 @@ insCond CodeGen::JumpKindToInsCond(emitJumpKind condition)
...
+
+        case GT_RSH:
+            return INS_OPTS_ASR;
+
+        case GT_RSZ:
+            return INS_OPTS_LSR;
+
+        default:
+            NO_WAY("expected a shift-op");
+            return INS_OPTS_NONE;
```
Comment (tannergooding) [inline src/coreclr/jit/codegenarm64.cpp:10338]: Should this stay `unreached()`?

File: src/coreclr/jit/lowerarmarch.cpp [hunk 2]
```diff
@@ -2264,7 +2264,30 @@ void Lowering::ContainCheckCast(GenTreeCast* node)
 //
 void Lowering::ContainCheckCompare(GenTreeOp* cmp)
 {
-    CheckImmedAndMakeContained(cmp, cmp->gtOp2);
+    GenTree* op1 = cmp->gtGetOp1();
+    GenTree* op2 = cmp->gtGetOp2();
+
+    if (CheckImmedAndMakeContained(cmp, op2))
+        return;
```
Comment (tannergooding) [inline src/coreclr/jit/lowerarmarch.cpp:2271]: You can also handle `op1` being an immediate and doing the relop swap here.

Something like:
```cpp
    if (CheckImmedAndMakeContained(cmp, op1))
    {
        std::swap(cmp->gtOp1, cmp->gtOp2);
        cmp->ChangeOper(cmp->SwapRelop(cmp->gtOper));
        return;
    }
```

**Review by tannergooding — APPROVED** [review]

File: src/coreclr/jit/codegenarm64.cpp (hunk 1 above)
Comment (TIHan) [reply src/coreclr/jit/codegenarm64.cpp:10338]: Shouldn't matter as long as something fails; also the `NO_WAY` is more descriptive. There are similar functions doing this.

**Review by BruceForstall — APPROVED** [review]
LGTM

**Pull request merged**